src/libs/response-rules-interpreter.ts

```diff
@@ -29,18 +29,24 @@ export class ResponseRulesInterpreter {
    * Choose the route response depending on the first fulfilled rule.
    * If no rule has been fulfilled get the first route response.
    */
-  public chooseResponse(): RouteResponse {
+  public chooseResponse(requestNumber: number): RouteResponse {
     if (this.randomResponse) {
       const randomStatus = Math.floor(
         Math.random() * this.routeResponses.length
       );
 
       return this.routeResponses[randomStatus];
     } else {
+      const maxRequestNumber = this.routeResponses
+      .map( routeResponse => routeResponse.rules)
+      .reduce((acc, val) => acc.concat(val), [])
+      .map( rule => Number.parseInt(rule.value, 10))
+      .reduce((a,b) => (a > b) ? a : b, 0);
```

Comment:
This could be simplified to avoid 4 iterations:

```suggestion
      .reduce((max, response) => Math.max(...[max, ...response.rules.flatMap(rule =>rule.target==='request_number' ?[parseInt(rule .value, 10)] : 0)]), 0)
```

Reply:
It is showing this error with this code TS2339: Property 'flatMap' does not exist on type 'ResponseRule[]'.

Reply:
It's a false alarm?

Reply:
ah no, it's not. We should then target es2019, only thing, I just saw that flatMap is only compatible with Node.js >=11. So for now we could revert to a reduce. 
But I still think you can get rid of the first `map` and last `reduce`, by using a `Math.max(...)` with spread operator.